maxun-core/src/interpret.ts

```
@@ -547,221 +547,225 @@ export default class Interpreter extends EventEmitter {
-          const currentHeight = await page.evaluate(() => document.body.scrollHeight);
-          if (currentHeight === previousHeight) {
-            const finalResults = await page.evaluate((cfg) => window.scrapeList(cfg), config);
-            allResults = allResults.concat(finalResults);
-            return allResults;
-          }
+        switch (config.pagination.type) {
+            case 'scrollDown':
+                await page.evaluate(() => window.scrollTo(0, document.body.scrollHeight));
+                await page.waitForTimeout(2000);
+
+                const currentHeight = await page.evaluate(() => document.body.scrollHeight);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)